doc/src/manual/arrays.md

```diff
@@ -479,6 +479,10 @@ If all the indices are scalars, then the result `X` is a single element from the
 `X` is an array with the same number of dimensions as the sum of the dimensionalities of all the
 indices.
 
+When some or all `I_k` are ranges or array of integers ("slicing"), the returned array would be a new array with
+copies of selected elements instead of sharing underlying memory with the original array `A`; to slice an array
+without making a copy, see [array views](@ref lib-arrays-views).
+
```

Comment:
I like the idea — this is good information to raise here.  Here's a slight re-working of your text:

```suggestion
indices. This returned array is a newly allocated container that contains the selected elements from the original array `A`.  As this uses separate storage, subsequent [indexed assignment](@ref man-indexed-assignment) into `X` does not change the elements of `A`.  To select a subset of an array without allocating new storage, see [array views](@ref lib-array-views).
```

I want to change a few things here:
* We don't make copies of elements, we just iteratively assign each selected element from `A` into the new storage in `X`
* I want to be pretty careful about "when" this happens — your list, for example, is missing `:`s.  Tying it into the previous section makes it a little simpler.

I do see how it could be helpful to use the words "slicing" and "slice" here... but I'm not quite sure how to best do that because we don't have a standard definition of what makes a slice so everyone brings their own context.  To me, for example, slicing doesn't include the case of an array of unsorted integers (like `I_1 = [3,1,4,1,5,9]`).  So I don't want to limit the descriptions here to imply that it only applies to slicing.

Reply:
well someone may search "slic*".

Our manual is very dense without covering the most useful operation in the simplest 1D case